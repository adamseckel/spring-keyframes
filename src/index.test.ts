import { spring } from './'

it('returns a keyframe array, with duplicate values removed', () => {
  const animation = spring({
    from: {
      opacity: 0,
      x: 0,
      scale: 0,
    },
    to: {
      opacity: 1,
      x: 100,
      scale: 1,
    },
  })

  expect(animation).toMatchSnapshot()
})

it('returns a keyframe array, with the passed unit', () => {
  const animation = spring(
    {
      from: {
        opacity: 0,
        x: 0,
      },
      to: {
        opacity: 1,
        x: 100,
      },
    },
    { unit: '%' }
  )

  expect(animation).toMatchSnapshot()
})

it('returns a keyframe array, with the desired precision', () => {
  const highPrecisionAnimation = spring(
    {
      from: {
        opacity: 0,
      },
      to: {
        opacity: 1,
      },
    },
    { precision: 4 }
  )

  expect(highPrecisionAnimation).toMatchSnapshot()

  const lowPrecisionAnimation = spring(
    {
      from: {
        opacity: 0,
      },
      to: {
        opacity: 1,
      },
    },
    { precision: 0 }
  )

  expect(lowPrecisionAnimation).toMatchSnapshot()
})
