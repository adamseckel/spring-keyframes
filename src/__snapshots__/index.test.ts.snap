// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a keyframe array, with duplicate values removed 1`] = `
Array [
  "0% {opacity: 0.01;transform: translateX(0.08px) scale3d(0.01, 0.01, 1);}",
  "1% {opacity: 0.02;transform: translateX(1.88px) scale3d(0.02, 0.02, 1);}",
  "2% {opacity: 0.06;transform: translateX(5.7px) scale3d(0.06, 0.06, 1);}",
  "3% {opacity: 0.12;transform: translateX(11.09px) scale3d(0.12, 0.12, 1);}",
  "4% {opacity: 0.18;transform: translateX(17.63px) scale3d(0.18, 0.18, 1);}",
  "5% {opacity: 0.24;transform: translateX(23.71px) scale3d(0.24, 0.24, 1);}",
  "6% {opacity: 0.32;transform: translateX(31.47px) scale3d(0.32, 0.32, 1);}",
  "7% {opacity: 0.4;transform: translateX(39.49px) scale3d(0.4, 0.4, 1);}",
  "8% {opacity: 0.48;transform: translateX(47.52px) scale3d(0.48, 0.48, 1);}",
  "9% {opacity: 0.56;transform: translateX(55.38px) scale3d(0.56, 0.56, 1);}",
  "10% {opacity: 0.62;transform: translateX(61.68px) scale3d(0.62, 0.62, 1);}",
  "11% {opacity: 0.69;transform: translateX(68.84px) scale3d(0.69, 0.69, 1);}",
  "12% {opacity: 0.76;transform: translateX(75.47px) scale3d(0.76, 0.76, 1);}",
  "13% {opacity: 0.82;transform: translateX(81.5px) scale3d(0.82, 0.82, 1);}",
  "14% {opacity: 0.87;transform: translateX(86.91px) scale3d(0.87, 0.87, 1);}",
  "15% {opacity: 0.91;transform: translateX(90.92px) scale3d(0.91, 0.91, 1);}",
  "16% {opacity: 0.96;transform: translateX(95.14px) scale3d(0.96, 0.96, 1);}",
  "17% {opacity: 0.99;transform: translateX(98.73px) scale3d(0.99, 0.99, 1);}",
  "18% {opacity: 1.02;transform: translateX(101.71px) scale3d(1.02, 1.02, 1);}",
  "19% {opacity: 1.05;transform: translateX(104.12px) scale3d(1.05, 1.05, 1);}",
  "20% {opacity: 1.06;transform: translateX(105.73px) scale3d(1.06, 1.06, 1);}",
  "21% {opacity: 1.08;transform: translateX(107.22px) scale3d(1.08, 1.08, 1);}",
  "22% {opacity: 1.09;transform: translateX(108.27px) scale3d(1.09, 1.09, 1);}",
  "23% {opacity: 1.09;transform: translateX(108.94px) scale3d(1.09, 1.09, 1);}",
  "24% {opacity: 1.1;transform: translateX(109.27px) scale3d(1.1, 1.1, 1);}",
  "25% {opacity: 1.1;transform: translateX(109.34px) scale3d(1.1, 1.1, 1);}",
  "26% {opacity: 1.1;transform: translateX(109.2px) scale3d(1.1, 1.1, 1);}",
  "27% {opacity: 1.09;transform: translateX(108.87px) scale3d(1.09, 1.09, 1);}",
  "28% {opacity: 1.09;transform: translateX(108.38px) scale3d(1.09, 1.09, 1);}",
  "29% {opacity: 1.08;transform: translateX(107.78px) scale3d(1.08, 1.08, 1);}",
  "30% {opacity: 1.08;transform: translateX(107.22px) scale3d(1.08, 1.08, 1);}",
  "31% {opacity: 1.07;transform: translateX(106.5px) scale3d(1.07, 1.07, 1);}",
  "32% {opacity: 1.06;transform: translateX(105.76px) scale3d(1.06, 1.06, 1);}",
  "33% {opacity: 1.06;transform: translateX(105.01px) scale3d(1.06, 1.06, 1);}",
  "34% {opacity: 1.05;transform: translateX(104.27px) scale3d(1.05, 1.05, 1);}",
  "35% {opacity: 1.04;transform: translateX(103.68px) scale3d(1.04, 1.04, 1);}",
  "36% {opacity: 1.03;transform: translateX(103px) scale3d(1.03, 1.03, 1);}",
  "37% {opacity: 1.03;transform: translateX(102.38px) scale3d(1.03, 1.03, 1);}",
  "38% {opacity: 1.02;transform: translateX(101.81px) scale3d(1.02, 1.02, 1);}",
  "39% {opacity: 1.02;transform: translateX(101.29px) scale3d(1.02, 1.02, 1);}",
  "40% {opacity: 1.01;transform: translateX(100.91px) scale3d(1.01, 1.01, 1);}",
  "41% {opacity: 1.01;transform: translateX(100.51px) scale3d(1.01, 1.01, 1);}",
  "42% {opacity: 1.01;transform: translateX(100.17px) scale3d(1.01, 1.01, 1);}",
  "43% {opacity: 1;transform: translateX(99.88px) scale3d(1, 1, 1);}",
  "44% {opacity: 1;transform: translateX(99.65px) scale3d(1, 1, 1);}",
  "45% {opacity: 1;transform: translateX(99.49px) scale3d(1, 1, 1);}",
  "46% {opacity: 1;transform: translateX(99.35px) scale3d(1, 1, 1);}",
  "47% {opacity: 1;transform: translateX(99.24px) scale3d(1, 1, 1);}",
  "48% {opacity: 1;transform: translateX(99.18px) scale3d(1, 1, 1);}",
  "49% {opacity: 1;transform: translateX(99.14px) scale3d(1, 1, 1);}",
  "50% {opacity: 1;transform: translateX(99.13px) scale3d(1, 1, 1);}",
  "51% {opacity: 1;transform: translateX(99.14px) scale3d(1, 1, 1);}",
  "52% {opacity: 1;transform: translateX(99.17px) scale3d(1, 1, 1);}",
  "53% {opacity: 1;transform: translateX(99.22px) scale3d(1, 1, 1);}",
  "54% {opacity: 1;transform: translateX(99.27px) scale3d(1, 1, 1);}",
  "55% {opacity: 1;transform: translateX(99.32px) scale3d(1, 1, 1);}",
  "56% {opacity: 1;transform: translateX(99.39px) scale3d(1, 1, 1);}",
  "57% {opacity: 1;transform: translateX(99.46px) scale3d(1, 1, 1);}",
  "58% {opacity: 1;transform: translateX(99.53px) scale3d(1, 1, 1);}",
  "59% {opacity: 1;transform: translateX(99.6px) scale3d(1, 1, 1);}",
  "60% {opacity: 1;transform: translateX(99.65px) scale3d(1, 1, 1);}",
  "61% {opacity: 1;transform: translateX(99.72px) scale3d(1, 1, 1);}",
  "62% {opacity: 1;transform: translateX(99.78px) scale3d(1, 1, 1);}",
  "63% {opacity: 1;transform: translateX(99.83px) scale3d(1, 1, 1);}",
  "64% {opacity: 1;transform: translateX(99.88px) scale3d(1, 1, 1);}",
  "65% {opacity: 1;transform: translateX(99.91px) scale3d(1, 1, 1);}",
  "66% {opacity: 1;transform: translateX(99.95px) scale3d(1, 1, 1);}",
  "67% {opacity: 1;transform: translateX(99.99px) scale3d(1, 1, 1);}",
  "68% {opacity: 1.01;transform: translateX(100.01px) scale3d(1.01, 1.01, 1);}",
  "69% {opacity: 1.01;transform: translateX(100.04px) scale3d(1.01, 1.01, 1);}",
  "70% {opacity: 1.01;transform: translateX(100.05px) scale3d(1.01, 1.01, 1);}",
  "71% {opacity: 1.01;transform: translateX(100.06px) scale3d(1.01, 1.01, 1);}",
  "72% {opacity: 1.01;transform: translateX(100.07px) scale3d(1.01, 1.01, 1);}",
  "73% {opacity: 1.01;transform: translateX(100.08px) scale3d(1.01, 1.01, 1);}",
  "74% {opacity: 1.01;transform: translateX(100.09px) scale3d(1.01, 1.01, 1);}",
  "77% {opacity: 1.01;transform: translateX(100.08px) scale3d(1.01, 1.01, 1);}",
  "79% {opacity: 1.01;transform: translateX(100.07px) scale3d(1.01, 1.01, 1);}",
  "81% {opacity: 1.01;transform: translateX(100.06px) scale3d(1.01, 1.01, 1);}",
  "83% {opacity: 1.01;transform: translateX(100.05px) scale3d(1.01, 1.01, 1);}",
  "84% {opacity: 1.01;transform: translateX(100.04px) scale3d(1.01, 1.01, 1);}",
  "86% {opacity: 1.01;transform: translateX(100.03px) scale3d(1.01, 1.01, 1);}",
  "88% {opacity: 1.01;transform: translateX(100.02px) scale3d(1.01, 1.01, 1);}",
  "90% {opacity: 1.01;transform: translateX(100.01px) scale3d(1.01, 1.01, 1);}",
  "93% {opacity: 1;transform: translateX(100px) scale3d(1, 1, 1);}",
  "100% {opacity: 1;transform: translateX(100px) scale3d(1, 1, 1);}",
]
`;

exports[`returns a keyframe array, with the desired precision 1`] = `
Array [
  "0% {opacity: 0.0008;}",
  "1% {opacity: 0.0188;}",
  "2% {opacity: 0.057;}",
  "3% {opacity: 0.1109;}",
  "4% {opacity: 0.1763;}",
  "5% {opacity: 0.2371;}",
  "6% {opacity: 0.3147;}",
  "7% {opacity: 0.3949;}",
  "8% {opacity: 0.4752;}",
  "9% {opacity: 0.5538;}",
  "10% {opacity: 0.6168;}",
  "11% {opacity: 0.6884;}",
  "12% {opacity: 0.7547;}",
  "13% {opacity: 0.815;}",
  "14% {opacity: 0.8691;}",
  "15% {opacity: 0.9092;}",
  "16% {opacity: 0.9514;}",
  "17% {opacity: 0.9873;}",
  "18% {opacity: 1.0171;}",
  "19% {opacity: 1.0412;}",
  "20% {opacity: 1.0573;}",
  "21% {opacity: 1.0722;}",
  "22% {opacity: 1.0827;}",
  "23% {opacity: 1.0894;}",
  "24% {opacity: 1.0927;}",
  "25% {opacity: 1.0934;}",
  "26% {opacity: 1.092;}",
  "27% {opacity: 1.0887;}",
  "28% {opacity: 1.0838;}",
  "29% {opacity: 1.0778;}",
  "30% {opacity: 1.0722;}",
  "31% {opacity: 1.065;}",
  "32% {opacity: 1.0576;}",
  "33% {opacity: 1.0501;}",
  "34% {opacity: 1.0427;}",
  "35% {opacity: 1.0368;}",
  "36% {opacity: 1.03;}",
  "37% {opacity: 1.0238;}",
  "38% {opacity: 1.0181;}",
  "39% {opacity: 1.0129;}",
  "40% {opacity: 1.0091;}",
  "41% {opacity: 1.0051;}",
  "42% {opacity: 1.0017;}",
  "43% {opacity: 0.9988;}",
  "44% {opacity: 0.9965;}",
  "45% {opacity: 0.9949;}",
  "46% {opacity: 0.9935;}",
  "47% {opacity: 0.9924;}",
  "48% {opacity: 0.9918;}",
  "49% {opacity: 0.9914;}",
  "50% {opacity: 0.9913;}",
  "51% {opacity: 0.9914;}",
  "52% {opacity: 0.9917;}",
  "53% {opacity: 0.9922;}",
  "54% {opacity: 0.9927;}",
  "55% {opacity: 0.9932;}",
  "56% {opacity: 0.9939;}",
  "57% {opacity: 0.9946;}",
  "58% {opacity: 0.9953;}",
  "59% {opacity: 0.996;}",
  "60% {opacity: 0.9965;}",
  "61% {opacity: 0.9972;}",
  "62% {opacity: 0.9978;}",
  "63% {opacity: 0.9983;}",
  "64% {opacity: 0.9988;}",
  "65% {opacity: 0.9991;}",
  "66% {opacity: 0.9995;}",
  "67% {opacity: 0.9999;}",
  "68% {opacity: 1.0001;}",
  "69% {opacity: 1.0004;}",
  "70% {opacity: 1.0005;}",
  "71% {opacity: 1.0006;}",
  "72% {opacity: 1.0007;}",
  "73% {opacity: 1.0008;}",
  "74% {opacity: 1.0009;}",
  "77% {opacity: 1.0008;}",
  "79% {opacity: 1.0007;}",
  "81% {opacity: 1.0006;}",
  "83% {opacity: 1.0005;}",
  "84% {opacity: 1.0004;}",
  "86% {opacity: 1.0003;}",
  "88% {opacity: 1.0002;}",
  "90% {opacity: 1.0001;}",
  "93% {opacity: 1;}",
  "100% {opacity: 1;}",
]
`;

exports[`returns a keyframe array, with the desired precision 2`] = `
Array [
  "0% {opacity: 1;}",
  "1% {opacity: 1;}",
  "18% {opacity: 2;}",
  "43% {opacity: 1;}",
  "68% {opacity: 2;}",
  "93% {opacity: 1;}",
  "100% {opacity: 1;}",
]
`;

exports[`returns a keyframe array, with the passed unit 1`] = `
Array [
  "0% {opacity: 0.01;transform: translateX(0.08%);}",
  "1% {opacity: 0.02;transform: translateX(1.88%);}",
  "2% {opacity: 0.06;transform: translateX(5.7%);}",
  "3% {opacity: 0.12;transform: translateX(11.09%);}",
  "4% {opacity: 0.18;transform: translateX(17.63%);}",
  "5% {opacity: 0.24;transform: translateX(23.71%);}",
  "6% {opacity: 0.32;transform: translateX(31.47%);}",
  "7% {opacity: 0.4;transform: translateX(39.49%);}",
  "8% {opacity: 0.48;transform: translateX(47.52%);}",
  "9% {opacity: 0.56;transform: translateX(55.38%);}",
  "10% {opacity: 0.62;transform: translateX(61.68%);}",
  "11% {opacity: 0.69;transform: translateX(68.84%);}",
  "12% {opacity: 0.76;transform: translateX(75.47%);}",
  "13% {opacity: 0.82;transform: translateX(81.5%);}",
  "14% {opacity: 0.87;transform: translateX(86.91%);}",
  "15% {opacity: 0.91;transform: translateX(90.92%);}",
  "16% {opacity: 0.96;transform: translateX(95.14%);}",
  "17% {opacity: 0.99;transform: translateX(98.73%);}",
  "18% {opacity: 1.02;transform: translateX(101.71%);}",
  "19% {opacity: 1.05;transform: translateX(104.12%);}",
  "20% {opacity: 1.06;transform: translateX(105.73%);}",
  "21% {opacity: 1.08;transform: translateX(107.22%);}",
  "22% {opacity: 1.09;transform: translateX(108.27%);}",
  "23% {opacity: 1.09;transform: translateX(108.94%);}",
  "24% {opacity: 1.1;transform: translateX(109.27%);}",
  "25% {opacity: 1.1;transform: translateX(109.34%);}",
  "26% {opacity: 1.1;transform: translateX(109.2%);}",
  "27% {opacity: 1.09;transform: translateX(108.87%);}",
  "28% {opacity: 1.09;transform: translateX(108.38%);}",
  "29% {opacity: 1.08;transform: translateX(107.78%);}",
  "30% {opacity: 1.08;transform: translateX(107.22%);}",
  "31% {opacity: 1.07;transform: translateX(106.5%);}",
  "32% {opacity: 1.06;transform: translateX(105.76%);}",
  "33% {opacity: 1.06;transform: translateX(105.01%);}",
  "34% {opacity: 1.05;transform: translateX(104.27%);}",
  "35% {opacity: 1.04;transform: translateX(103.68%);}",
  "36% {opacity: 1.03;transform: translateX(103%);}",
  "37% {opacity: 1.03;transform: translateX(102.38%);}",
  "38% {opacity: 1.02;transform: translateX(101.81%);}",
  "39% {opacity: 1.02;transform: translateX(101.29%);}",
  "40% {opacity: 1.01;transform: translateX(100.91%);}",
  "41% {opacity: 1.01;transform: translateX(100.51%);}",
  "42% {opacity: 1.01;transform: translateX(100.17%);}",
  "43% {opacity: 1;transform: translateX(99.88%);}",
  "44% {opacity: 1;transform: translateX(99.65%);}",
  "45% {opacity: 1;transform: translateX(99.49%);}",
  "46% {opacity: 1;transform: translateX(99.35%);}",
  "47% {opacity: 1;transform: translateX(99.24%);}",
  "48% {opacity: 1;transform: translateX(99.18%);}",
  "49% {opacity: 1;transform: translateX(99.14%);}",
  "50% {opacity: 1;transform: translateX(99.13%);}",
  "51% {opacity: 1;transform: translateX(99.14%);}",
  "52% {opacity: 1;transform: translateX(99.17%);}",
  "53% {opacity: 1;transform: translateX(99.22%);}",
  "54% {opacity: 1;transform: translateX(99.27%);}",
  "55% {opacity: 1;transform: translateX(99.32%);}",
  "56% {opacity: 1;transform: translateX(99.39%);}",
  "57% {opacity: 1;transform: translateX(99.46%);}",
  "58% {opacity: 1;transform: translateX(99.53%);}",
  "59% {opacity: 1;transform: translateX(99.6%);}",
  "60% {opacity: 1;transform: translateX(99.65%);}",
  "61% {opacity: 1;transform: translateX(99.72%);}",
  "62% {opacity: 1;transform: translateX(99.78%);}",
  "63% {opacity: 1;transform: translateX(99.83%);}",
  "64% {opacity: 1;transform: translateX(99.88%);}",
  "65% {opacity: 1;transform: translateX(99.91%);}",
  "66% {opacity: 1;transform: translateX(99.95%);}",
  "67% {opacity: 1;transform: translateX(99.99%);}",
  "68% {opacity: 1.01;transform: translateX(100.01%);}",
  "69% {opacity: 1.01;transform: translateX(100.04%);}",
  "70% {opacity: 1.01;transform: translateX(100.05%);}",
  "71% {opacity: 1.01;transform: translateX(100.06%);}",
  "72% {opacity: 1.01;transform: translateX(100.07%);}",
  "73% {opacity: 1.01;transform: translateX(100.08%);}",
  "74% {opacity: 1.01;transform: translateX(100.09%);}",
  "77% {opacity: 1.01;transform: translateX(100.08%);}",
  "79% {opacity: 1.01;transform: translateX(100.07%);}",
  "81% {opacity: 1.01;transform: translateX(100.06%);}",
  "83% {opacity: 1.01;transform: translateX(100.05%);}",
  "84% {opacity: 1.01;transform: translateX(100.04%);}",
  "86% {opacity: 1.01;transform: translateX(100.03%);}",
  "88% {opacity: 1.01;transform: translateX(100.02%);}",
  "90% {opacity: 1.01;transform: translateX(100.01%);}",
  "93% {opacity: 1;transform: translateX(100%);}",
  "100% {opacity: 1;transform: translateX(100%);}",
]
`;
